linetools/spectra/xspectrum1d.py

```diff
@@ -84,92 +75,205 @@ def from_file(self, ifile, **kwargs):
+
         # Generate
-        if len(ltuple) == 2:  # wave, flux
-            spec = cls.from_array(uwave, u.Quantity(ltuple[1]), **kwargs)
-            spec.uncertainty = None
-            spec.co = None
-        elif len(ltuple) == 3:
-            spec = cls.from_array(uwave, u.Quantity(ltuple[1]),
-                uncertainty=StdDevUncertainty(ltuple[2]), **kwargs)
-            spec.co = None
-        else:
-            spec = cls.from_array(uwave, u.Quantity(ltuple[1]),
-                uncertainty=StdDevUncertainty(ltuple[2]), **kwargs)
-            spec.co = ltuple[3]
+        spec = cls(iwave, iflux, sig=ltuple[2], co=ltuple[3], units=dict(wave=wv_unit, flux=fx_unit), **kwargs)
 
-        spec.filename = 'none'
         # Return
         return spec
 
+    def __init__(self, wave, flux, sig=None, co=None, units=None, select=0,
+                 meta=None, verbose=True):
+        """
+        Parameters
+        ----------
+        wave : ndarray  [if 2D, the first axis is nspec]
+        flux : ndarray
+        sig : ndarray, optional
+        units : dict, optional
+          Dict containing the units of wavelength, flux
+          Required keys are 'wave' and 'flux'
+        select : int, optional
+          Selected Spectrum
+        """
+        # Error checking
+        if not isinstance(wave, np.ndarray):
+            raise IOError("Input x-vector must be an ndarray")
+        if not isinstance(flux, np.ndarray):
+            raise IOError("Input y-vector must be an ndarray")
+        if wave.shape[0] != flux.shape[0]:
+            raise IOError("Shape of x and y vectors must be identical")
+
+        # Handle many spectra
+        if len(wave.shape) == 1:
+            self.nspec = 1
+            self.npix = wave.shape[0]
+        else:
+            self.nspec = wave.shape[0]
+            self.npix = wave.shape[1]
+        self.select = select
+
+        if verbose:
+            print("We have {:d} spectra with {:d} pixels each.".format(
+                self.nspec, self.npix))
+
+        # Data array
+        self._data = np.empty((self.nspec,), #self.npix),
+                               dtype=[(str('wave'), 'float64', (self.npix)),
+                                      (str('flux'), 'float32', (self.npix)),
+                                      (str('sig'),  'float32', (self.npix)),
+                                      (str('co'),   'float32', (self.npix)),
+                                     ])
+        self._data['wave'] = np.reshape(wave, (self.nspec, self.npix))
+        self._data['flux'] = np.reshape(flux, (self.nspec, self.npix))
+        if sig is not None:
+            self._data['sig'] = np.reshape(sig, (self.nspec, self.npix))
+            self.flag_sig = True
+        else:
+            self.flag_sig = False
+        if co is not None:
+            self._data['co'] = np.reshape(co, (self.nspec, self.npix))
+            self.flag_co = True
+        else:
+            self._data['co'] = 1.
```

Comment:
Wouldn't it be better to set `self._data['co'] = None` if continuum is not defined?